addons/knobs/src/components/types/Array.tsx

```diff
@@ -2,18 +2,16 @@ import PropTypes from 'prop-types';
 import React, { ChangeEvent, Component, WeakValidationMap } from 'react';
 
 import { Form } from '@storybook/components';
+import { KnobControlConfig, KnobControlProps } from './types';
 
-type ArrayTypeKnobValue = string[];
+export type ArrayTypeKnobValue = string[] | readonly string[];
```

Comment:
Is this breaking type change intentional? Using `number`s will cause a typescript error.

Reply:
To quote the README for this add-on:

> array
> Allows you to get an array of strings from the user.

Unfortunately, the package from DefinitelyTyped is inaccurate—that field uses a text-based input of comma delimited values, so even if you were to pass a number array initially, it would subsequently return a string array on change, i.e., here's no number parsing logic for that control.

That said, I'd love to hear more about your use case and help in any way I can. Would you mind sharing?

Reply:
Hey @emilio-martinez Thanks for the reply. Yeah it seems like you guys are just having to bite the bullet on these type changes going from javascript to typescript, which I totally feel your pain.

This is my use case 👇. It's fixable by just parsing the values and there is probably a better way to go about it. Just got this error when upgrading from `v5.1.9` to `v5.2.0-beta.19`.
https://github.com/elastic/elastic-charts/blob/8ebe600df1ae476e25032ab1c7707de398143e43/stories/annotations.tsx#L41

I'm curious if you guys are gradually upgrading you codebase to ts or doing it in a single release (I'm too lazy to check for myself)? If it is in a single release why wouldn’t you just release ts in `v6` rather than producing all these stricter type errors?

Thanks 👍

Reply:
TS migration is an epic started at the end of 2018, we recently reached an important step as 50% of Storybook codebase in TS 🎉.
Migration progress is followed in https://github.com/storybookjs/storybook/issues/5030 😉. Avoiding breaking changes is one of our goals but we somethings found weird/wrong types/docs even in SB own codebase.

Reply:
@gaetanmaisse Sweet congrats on the big 5-0!! 🎉 All the best keep up the great work.

Reply:
Actually, there are many work around and we don't buy in strict type and `noImplicitAny` at day one.  For that reason, there are sheer amount of `any` flood in our typing. That somehow explains why we inevitably will have breaking changes in type system, we need to improve it gradually due to the fact that the codebase is pretty huge.